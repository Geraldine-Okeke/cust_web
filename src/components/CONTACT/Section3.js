import React, { useState } from 'react';

const Section3 = () => {
  const [formErrors, setFormErrors] = useState({});

  const handleSubmit = async (e) => {
    e.preventDefault();

    const formData = {
      firstName: e.target.elements.fn.value,
      lastName: e.target.elements.ln.value,
      email: e.target.elements.em.value,
      phoneNumber: e.target.elements.pn.value,
      message: e.target.elements.ta.value,
    };

    
    const errors = {};
    for (const field in formData) {
      if (!formData[field]) {
        errors[field] = `${field.replace(/([A-Z])/g, ' $1').trim()} is required`;
      }
    }

    if (Object.keys(errors).length > 0) {
      setFormErrors(errors);
      return;
    }

    console.log(formData);

    try {
      const response = await fetch('', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(formData),
      });

      if (response.ok) {
        console.log('Form data submitted successfully!');
        e.target.reset();
        setFormErrors({});
      } else {
        console.error('Form data submission failed.');
      }
    } catch (error) {
      console.error('An error occurred:', error);
    }
  };

  return (
    <>
      <h1 className="font-bold text-3xl text-center mt-10 mb-20" id="contact">SEND US A MESSAGE</h1>
      <form className="" onSubmit={handleSubmit}>
        <h1 className="text-3xl font-bold mb-5 pl-4">Name</h1>
       <div className="md:flex p-4 gap-10">
        <div className=" md:w-1/2  ">
          <div className="flex flex-col">
            <input type="text" className="p-2 border rounded" id="fn" />
            <label htmlFor="fn" className="mb-2 ">First Name</label>
          </div>
        </div>
          <div className=" md:w-1/2 ">
            <div className="flex flex-col">
              <input type="text" className="p-2 border rounded" id="ln"  />
              <label htmlFor="ln" className="mb-2 ">Last Name</label>
            </div>
          </div>
       </div>
        
       <div className="md:flex p-4 gap-9">
        
        <div className=" md:w-1/2  ">
        <h1 className="text-3xl font-bold mb-5">Email</h1>
          <div className="flex flex-col">
            <input type="email" className="p-2 border rounded" id="em"  />
            <label htmlFor="em" className="mb-2 ">example@example.com</label>
          </div>
        </div>
          <div className=" md:w-1/2 ">
          <h1 className="text-3xl font-bold mb-5">Phone Number</h1>
            <div className="flex flex-col">
              <input type="tel" className="p-2 border rounded" id="pn" />
              <label htmlFor="pn" className="mb-2 ">Please enter a valid phone number</label>
            </div>
          </div>
       </div>
        <textarea id="ta" className="w-full p-4 border rounded-md resize-y focus:outline-none focus:ring focus:border-blue-300" rows="4" placeholder="Enter your message"></textarea>
        <label htmlFor="ta"className='pl-4' >Message</label> <br/>
        <button
          type="submit"
          className={`px-4 py-2 float-right my-8 mr-10 bg-blue-900 w-32 font-bold text-white rounded-full hover:bg-blue-600 focus:outline-none focus:ring focus:border-blue-300 ${
            Object.keys(formErrors).length > 0 ? 'opacity-50 cursor-not-allowed' : ''
          }`}
          disabled={Object.keys(formErrors).length > 0}
        >
          SEND
        </button>

      </form>


    </>
  )
  
  
}
export default Section3