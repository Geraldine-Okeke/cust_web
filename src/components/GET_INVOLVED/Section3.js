import { Link } from 'react-router-dom';
export default function Section3(){
  return(
    <>
      <div className="flex flex-wrap backG text-white rounded-t-xl h-auto items-center">
        <div className="w-full md:w-1/3 p-4 text-center">
          <div className='flex justify-center py-10'>
            <Link to="/GET_INVOLVED/Volunteer">
              <button className='bg-yellow-500 text-black rounded-full w-auto px-10 py-3 font-bold mx-auto'>VOLUNTEER</button>
            </Link>
          </div>
          
          <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor 
          incididunt ut labore et dolore magna aliqua. Ut enim ad
          </p>
        </div>
        <div className="w-full md:w-1/3 p-4 text-center">
          <div className='flex justify-center py-10'>
              <Link to="/GET_INVOLVED/Sponsor">
                <button className='bg-yellow-500 text-black rounded-full w-auto px-10 py-3 font-bold mx-auto'>SPONSOR</button>
              </Link>
            </div>
            
            <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor 
            incididunt ut labore et dolore magna aliqua. Ut enim ad
            </p>
          </div>
        <div className="w-full md:w-1/3 p-4 text-center">
          <div className='flex justify-center py-10'>
              <Link to="/GET_INVOLVED/Mentor">
                <button className='bg-yellow-500 text-black rounded-full w-auto px-10 py-3 font-bold mx-auto'>MENTOR</button>
              </Link>
            </div>
            
            <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor 
            incididunt ut labore et dolore magna aliqua. Ut enim ad
            </p>
          </div>
      </div>

    </>
  )
}