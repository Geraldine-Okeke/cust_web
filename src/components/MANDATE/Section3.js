import React from 'react';
import team from './Teams';
import  { useEffect, useRef } from 'react';
export default function Section3() {
  const teamRef = useRef(null); 

  useEffect(() => {
    
    const locationHash = window.location.hash;
    if (locationHash === '#team') {
      
      teamRef.current.scrollIntoView({ behavior: 'smooth' });
    }
  }, []);
  return (
    <>
    <div ref={teamRef} id='team'>
    <h1 className='font-extrabold text-3xl text-center' >Our Team</h1>
    </div>
    <div className="md:p-20 grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-4 lg:w-full">
      {team.map((employee, index) => (
        <div key={index} className=" p-10 flex flex-col items-center">
          <img src={employee.image} alt={employee.name} className="w-full  " />
          <div className=' blue h-10 w-full'><h2 className='font-bold text-2xl text-center'>{employee.name}</h2></div>
          <div className='shadow-xl' >
            <span className="text-gray-600 font-semibold">{employee.contact}</span>
            <p>{employee.summary}</p>
          </div>
          
        </div>
      ))}
    </div>
    </>
  );
}

      