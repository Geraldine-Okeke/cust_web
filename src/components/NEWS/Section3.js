import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import Events from './Events';

export default function Section3() {
  const [events, setEvents] = useState([]);
  const [useApi, setUseApi] = useState(false); // I will Set this to true when API is ready

  useEffect(() => {
    if (useApi) {
      // Fetch events from API and update the events state
      async function fetchEvents() {
        try {
          const response = await fetch('api endpoint');
          const data = await response.json();
          setEvents(data);
        } catch (error) {
          console.error('Error fetching events from API:', error);
        }
      }

      fetchEvents();
    } else {
      setEvents(Events); // Use local events from Events.js
    }
  }, [useApi]);

  return (
    <>
      {events.map((event, index) => (
      <div className='flex flex-col md:flex-row border p-2 md:p-12 rounded-xl shadow-lg m-6' key={index}>
        <div className="md:w-1/2 p-4 relative">
          <img src={event.image} alt={`Event ${index + 1}`} className="w-full h-auto md:w-3/4 rounded-xl" />
          <span className="text-black absolute top-1/2 left-0 transform -translate-y-1/2 h-10 flex px-2 rounded-lg items-center text-center font-bold yellow">{event.date}</span>
        </div>
        <div className="md:w-1/2 p-4">
          <h1 className='text-4xl font-bold'>About this Event</h1>
          <p className='leading-8 pt-4 md:pt-16'>{event.description}</p>
          <Link to={event.link}>
            <button className='yellow text-black font-bold mt-4 rounded-full px-6 py-2 hover:bg-yellow-700'>JOIN NOW</button>
          </Link>
        </div>
      </div>
      
      ))}
    </>
  );
}
