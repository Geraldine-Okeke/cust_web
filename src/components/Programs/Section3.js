const Section3 = ({ programs, heading, backgroundC , textColor}) => {
  
  return (
    <>
    <div  style={{ backgroundColor: backgroundC, color : textColor }} className="mb-0 pb-10" >
      <div className="md:p-10" ><h1 className=" mb-10 text-center font-extrabold text-3xl" id='All_Progs'  >{heading}</h1></div>
        <div className="grid grid-cols-1 md:grid-cols-2 gap-4">
        {programs.map((program, index) => (
          <div key={index} className="p-4 md:p-0 md:flex  rounded-xl" style={{ backgroundColor: program.bg_color }}>
            <div className="md:w-1/2" id={program.id}>
              <img src={program.image} alt="prog" className="w-full md:h-52 rounded-lg pt-3 pl-3" />
              <h2 className="text-center font-bold" >{program.title.toUpperCase()}</h2>
              <hr className="border-t-4 mx-auto w-1/2 border-yellow-400"/>
            </div>
            <div className="md:w-1/2 pl-4">
              <div  className="mt-4">{program.details}</div>
            </div>
          </div>
        ))}
      </div>
    </div>
      
    </>
    
  );
};

export default Section3;
