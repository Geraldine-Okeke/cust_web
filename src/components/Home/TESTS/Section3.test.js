import React from 'react';
import renderer from 'react-test-renderer';
import Section3 from '../Section3'; 

test('Section3 component snapshot', () => {
  const component = renderer.create(<Section3 />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
