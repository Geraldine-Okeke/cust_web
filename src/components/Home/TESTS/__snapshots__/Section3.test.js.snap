// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section3 component snapshot 1`] = `
<div
  className=" bg-custom-teal w-full   h-auto relative rounded-xl"
>
  <div
    className="  mt-10"
  >
    <h2
      className="font-bold text-3xl text-white pt-4 sm:ml-20 w-full "
    >
      OUR VALUE ADDING ROADMAP 
      <br />
       (HESS MANDATE)
    </h2>
    <hr
      className="  font-extrabold border-t-4 bg-custom-yellow mb-4 w-60 sm:ml-20"
    />
  </div>
  <img
    alt="logo"
    className="absolute -right-20  top-3 hidden md:block "
    src="Section2.png"
  />
  <div
    className=" flex-row md:pl-20 h-full  md:pr-20 py-16 grid md:flex md:flex-row\\""
  >
    <div
      className="w-full mt-20 -mr-10 z-10 bg-gray-800 text-white justify-start par h-auto  pb-3 md:w-1/2 lg:w-1/2"
    >
      <h3
        className="font-bold mb-5 pt-10 px-10"
      >
        Healthy communities:
      </h3>
      <p
        className="px-10 "
      >
         We believe when our communities are healthy and Eco-friendly, people thrive. Consequently, often many of the services that keep people healthy are difficult to access for the people who need them most. We will facilitate enabling environments to promote creativity, innovation and collaboration for healthcare causes and objectives. We will ensure that through our community networks every Hungry Citizen within our reach gets Food Assistance and the Homeless get a clean living environment with access to disease prevention medicine.
      </p>
    </div>
    <div
      className="   md:w-1/2 mt-0  "
    >
      <img
        alt="people"
        className="md:w-full  people"
        src="Section3.png"
      />
    </div>
  </div>
</div>
`;
